src/movies.js

```diff
@@ -1,22 +1,78 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors(moviesArray) {}
+
+function getAllDirectors(moviesArray) {
+    const movie = moviesArray.map((movie) => movie.director)
+    return movie
+}
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies(moviesArray) {}
+function howManyMovies(moviesArray) {
+
+   
+    if (moviesArray.length === 0) return 0;
+
+   const genreMovie = moviesArray.filter(function(movie) {
+    if (movie.genre.includes("Drama") & movie.director === "Steven Spielberg") {
+     return true;
```

Comment:
`.filter()` is an `array` `method` that already `returns` you `true` if the `element` passes the `condition`.
So here you can just do:

```
function howManyMovies(moviesArray) {
  if (moviesArray.length === 0) return 0;

  const genreMovie = moviesArray.filter(
    (movie) => movie.genre.includes("Drama") && movie.director === "Steven Spielberg");
  return genreMovie.length;
}
```

There's no need for the `if` neither the `return true`
Also you're missing one `&` in the `and (&&)` of the condition